test/runtests.jl

```diff
@@ -406,6 +410,68 @@ function test_potential_energy_equation_terms(model; geopotential_height = nothi
         end
     end
 
+    Eb = BackgroundPotentialEnergy(model; geopotential_height)
+
+    Eb_field = Field(Eb)
+    @test Eb isa AbstractOperation
+    @test Eb_field isa Field
+    compute!(Eb_field)
+
+    return nothing
+end
+function test_1D_reference_field(model; geopotential_height = 0)
+
+    if model.buoyancy isa BuoyancyTracerModel
+
+        b = reshape(interior(model.tracers.b), :)
+        p = sortperm(b, rev = true)
+        b_sorted = b[p]
+
+        v_grid = VolumeField(model.grid)
+        A = sum(AreaField(model.grid))
+        z✶_check = -cumsum(reshape(v_grid, :)[p]) / A
+
+        sorted_buoyancy, z✶ = OneDReferenceField(model.tracers.b, rev = true)
+
+        @test isequal(reshape(interior(sorted_buoyancy), :), b_sorted)
+        @test isequal(reshape(interior(z✶), :), z✶_check)
+
+    elseif model.buoyancy isa BuoyancyLinearEOSModel
+
+        b = Field(linear_eos_buoyancy(model.grid, model.buoyancy.model, model.tracers))
+        compute!(b)
+        b_flat = reshape(interior(b), :)
+        p = sortperm(b_flat, rev = true)
+        b_sorted = b_flat[p]
+
+        v_grid = VolumeField(model.grid)
+        A = sum(AreaField(model.grid))
+        z✶_check = -cumsum(reshape(v_grid, :)[p]) / A
+
+        sorted_buoyancy, z✶ = OneDReferenceField(b, rev = true)
+
+        @test isequal(reshape(interior(sorted_buoyancy), :), b_sorted)
+        @test isequal(reshape(interior(z✶), :), z✶_check)
+
+    elseif model.buoyancy isa BuoyancyBoussinesqEOSModel
+
+        ρ = Field(seawater_density(model; geopotential_height))
+        compute!(ρ)
+        ρ_flat = reshape(interior(ρ), :)
+        p = sortperm(ρ_flat)
+        ρ_sorted = ρ_flat[p]
+
+        v_grid = VolumeField(model.grid)
+        A = sum(AreaField(model.grid))
+        z✶_check = -cumsum(reshape(v_grid, :)[p]) / A
+
+        sorted_density, z✶ = OneDReferenceField(ρ)
+
+        @test isequal(reshape(interior(sorted_density), :), ρ_sorted)
+        @test isequal(reshape(interior(z✶), :), z✶_check)
```

Comment:
The comment above extends to these other `model.buoyancy` types. A separate comment here is that you're writing a lot of code 3 times. You can probably reuse code, so that you write most things only once.